aws/cloudformation/components/cache.yml.erb

```diff
@@ -36,3 +33,59 @@
       AZMode: cross-az
       PreferredAvailabilityZones: <%= AVAILABILITY_ZONES.first(2).to_json %>
       VpcSecurityGroupIds: [!ImportValue VPC-MemcachedSecurityGroup]
+  
+  MemcachedParameterGroup:
+    Type: AWS::ElastiCache::ParameterGroup
+    Properties:
+      CacheParameterGroupFamily: memcached1.4
+      Description: Memcached modified max_item_size.
+      Properties:
+        max_item_size: <%=64.megabytes%>
+
+  MemcachedCPUUsageAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: Production Shared Cache CPU Utilization
+      AlarmDescription: The Elasticache Memcached cluster, which we use to support
+          CDO.shared_cache in production is working much harder than usual. May
+          impact site performance if utilization is too high, but as of Jan 2025
+          we are unsure what the actual impact would be, and how high exactly this
+          alarm threshold should be set.
+      ActionsEnabled: true
+      AlarmActions:
+          - !Ref InfraUrgent
+      MetricName: CPUUtilization
+      Namespace: AWS/ElastiCache
+      Statistic: Average
+      Dimensions:
+          - Name: CacheClusterId
+            Value: !GetAtt Memcached.Id
+      Period: 300
+      EvaluationPeriods: 5
+      DatapointsToAlarm: 5
+      Threshold: 20
+      ComparisonOperator: GreaterThanThreshold
+      TreatMissingData: breaching
+
+  MemcachedCPUUsageWarningAlarm
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: Production Shared Cache CPU Utilization - WARN
+      AlarmDescription: The Elasticache Memcached cluster, which supports
+          CDO.shared_cache in production is working a lot harder than usual.
+          Unlikely to reflect actual user impact, but should be investigated.
+      ActionsEnabled: true
+      AlarmActions:
+          - !Ref !InfraWorkdayUrgent
```

Comment:
Can we use conditional logic so that this is only workday urgent for production? Eventually our test system and other systems will have parity with production and will provision memcached for the shared cache.

Reply:
shoot, yep, need to do some environment switching here.

Reply:
This is inside an `if frontends`, but regardless these are also per-stack SNS topics, which means that "staging", "test", and "autoscale-prod" have their own "InfraUrgent" topics, but we've only configured PagerDuty alerts for production. We may choose to trigger pagerduty for alerts in other environments, but we could choose, for example, for InfraUrgent in Test to only page during work hours.